src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -2280,10 +2280,16 @@ public async Task<JArray> GetValuesFromDebuggerProxyAttribute(SessionId sessionI
                             var assemblyNameArg = await GetFullAssemblyName(sessionId, assemblyIdArg, token);
                             var classNameArg = await GetTypeNameOriginal(sessionId, genericTypeArgs[k], token);
                             typeToSearch += classNameArg +", " + assemblyNameArg;
+                            if (k + 1 < genericTypeArgs.Count)
+                                typeToSearch += "], [";
+                            else
+                                typeToSearch += "]";
                         }
-                        typeToSearch += "]]";
+                        typeToSearch += "]";
                         typeToSearch +=  ", " + assemblyName;
                         var genericTypeId = await GetTypeByName(sessionId, typeToSearch, token);
+                        if (genericTypeId < 0)
+                            return null;
                         methodId = await GetMethodIdByName(sessionId, genericTypeId, ".ctor", token);
```

Comment:
Won't this fail to find the correct .ctor, if there is more than one on the type?

Also, `DebuggerTypeProxy` accepts a string arg with the type name. That doesn't have to be in this PR though.